### PR TITLE
Feature: Check network connection

### PR DESCRIPTION
**Pull request by Cal-Hagner**
Added a function to determine if the user is connected to the Internet (for the purposes of downloading/caching). This is a step towards full download functionality.

Tested this locally, and it works, I am unsure of how to write a unit test for it, as validating the network connection would mean checking it via a different method and validating the two methods.

Any suggestions for unit testing?

### CODE REVIEW

**Comment by Cal-Hagner** [comment]
@rishiosaur @cjdenio Any thoughts?

**Comment by cjdenio** [comment]
Looks good, but first, did you add `http2` as a dependency? And second, should we use a non-deprecated package?

**Comment by Cal-Hagner** [comment]
...I'm not sure what you mean. I thought that `http2` was [part of the standard Node API with `Stable` stability](https://nodejs.org/api/http2.html#http2_http_2).

**Comment by cjdenio** [comment]
Whoops, sorry! I thought you were using some crazy [npm package](https://npmjs.com/http2). :joy: 
Looks all good!

File: src/struct/package.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "type": "module"
```
[the remark this answers is not in the document]
Comment (Cal-Hagner) [reply src/struct/package.json:2]: ES6 imports are better and are supported in node >= v13, so I thought we should use them. For some reason, I couldn't import the `src/struct/index.ts` into other files (for manual testing) until I added that thing to declare it as a module so ES6 module import/exports could be used.

**Comment by Cal-Hagner** [comment]
@cjdenio I can just leave everything in normal node `require` for now, I think it's more important to just get this functionality into the repo so that we can build on top of it. We can discuss ES6 modules later down the road, beyond this first milestone.

**Comment by Cal-Hagner** [comment]
Okay, I'll switch back to CommonJS modules and then push and merge.

**Comment by Matt-Gleich** [comment]
You should add unit tests before merging. Just run the function and make sure it returns true.

**Pull request merged**